lambda-parent/lambda/src/test/java/org/wicketstuff/lambda/model/LambdaModelTest.java

```diff
@@ -0,0 +1,87 @@
+package org.wicketstuff.lambda.model;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.Model;
+import org.junit.Test;
+import org.wicketstuff.lambda.OptionalFunction;
+
+public class LambdaModelTest {
+	
+	@Test
+	public void testGetter() {
+		Person person = new Person();
+		person.setName("A");
+		IModel<String> nameModel = new LambdaModel<>(Model.of(person), Person::getName);
+		assertEquals("A", nameModel.getObject());
+	}
+
+	@Test(expected = NullPointerException.class)
+	public void testUnhandledNullModelObject() {
+		IModel<String> nameModel = new LambdaModel<>(Model.of((Person)null), Person::getName);
+		assertEquals("A", nameModel.getObject());
```

Comment:
Fixed.

Thanks
Andrew

On Wed, Sep 30, 2015 at 4:56 PM, Martin Grigorov <notifications@github.com>
wrote:

> In
> lambda-parent/lambda/src/test/java/org/wicketstuff/lambda/model/LambdaModelTest.java
> <https://github.com/wicketstuff/core/pull/421#discussion_r40850498>:
>
> > +import org.wicketstuff.lambda.OptionalFunction;
> > +
> > +public class LambdaModelTest {
> > +	
> > +	@Test
> > +	public void testGetter() {
> > +		Person person = new Person();
> > +		person.setName("A");
> > +		IModel<String> nameModel = new LambdaModel<>(Model.of(person), Person::getName);
> > +		assertEquals("A", nameModel.getObject());
> > +	}
> > +
> > +	@Test(expected = NullPointerException.class)
> > +	public void testUnhandledNullModelObject() {
> > +		IModel<String> nameModel = new LambdaModel<>(Model.of((Person)null), Person::getName);
> > +		assertEquals("A", nameModel.getObject());
>
> Please simplify this just to nameModel.getObject() because it is a bit
> confusing at first sight.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wicketstuff/core/pull/421/files#r40850498>.
>